pr_agent/tools/pr_extended_code_suggestions.py

```diff
@@ -0,0 +1,215 @@
+from typing import List
+import copy
+import json
+import logging
+import textwrap
+from typing import Dict, Any
+
+import yaml
+from jinja2 import Environment, StrictUndefined
+
+from pr_agent.algo.ai_handler import AiHandler
+from pr_agent.algo.pr_processing import get_pr_multi_diffs, retry_with_fallback_models
+from pr_agent.algo.token_handler import TokenHandler
+from pr_agent.algo.utils import try_fix_json
+from pr_agent.config_loader import get_settings
+from pr_agent.git_providers import get_git_provider
+from pr_agent.git_providers.git_provider import get_main_pr_language
+
+
+class PRExtendedCodeSuggestions:
+    def __init__(self, pr_url: str, cli_mode=False, args: list = None):
+
+        self.git_provider = get_git_provider()(pr_url)
+        self.main_language = get_main_pr_language(
+            self.git_provider.get_languages(), self.git_provider.get_files()
+        )
+
+        self.ai_handler = AiHandler()
+        self.patches_diff = None
+        self.prediction = None
+        self.cli_mode = cli_mode
+        self.vars = {
+            "title": self.git_provider.pr.title,
+            "branch": self.git_provider.get_pr_branch(),
+            "description": self.git_provider.get_pr_description(),
+            "language": self.main_language,
+            "diff": "",  # empty diff for initial calculation
+            "num_code_suggestions": get_settings().pr_extendeted_code_suggestions.num_code_suggestions_per_chunk,
+            "extra_instructions": get_settings().pr_extendeted_code_suggestions.extra_instructions,
+            "commit_messages_str": self.git_provider.get_commit_messages(),
+        }
+        self.token_handler = TokenHandler(self.git_provider.pr,
+                                          self.vars,
+                                          get_settings().pr_code_suggestions_prompt.system,
+                                          get_settings().pr_code_suggestions_prompt.user)
+
+    async def run(self):
```

Comment:
**Suggestion:** Consider adding a return type hint to the 'run' function to improve code readability and maintainability.
```suggestion
    async def run(self) -> None:
```